### PR TITLE
Move signaling to vue buttons

### PR DESCRIPTION
**Pull request by nickvergessen**

### CODE REVIEW

**Comment by nickvergessen** [comment]
Removed the checkbox change (is now tracked in #7707 ) so all buttons are done with this PR

File: src/components/AdminSettings/SignalingServer.vue [hunk 1]
```diff
@@ -38,17 +38,23 @@
...
-		<a v-show="!loading"
-			v-tooltip.auto="t('spreed', 'Delete this server')"
-			class="icon icon-delete"
-			@click="removeServer" />
+		<Button v-show="!loading"
+			type="tertiary-no-background"
+			:aria-label="t('spreed', 'Delete this server')"
+			@click="removeServer">
+			<template #icon>
+				<Delete />
```
Comment (raimund-schluessler) [inline src/components/AdminSettings/SignalingServer.vue:46]: Unless you really want the icon to be the default size (`24`), you should probably set the size we agreed on kind of globally, which is `20`:
```suggestion
				<Delete :size="20"/>
```
see e.g. https://github.com/nextcloud/nextcloud-vue/blob/71468d3a42bc0e53899e31e1df6fdfb093ec3840/src/components/AppNavigationToggle/AppNavigationToggle.vue#L33

File: src/components/AdminSettings/SignalingServers.vue [hunk 2]
```diff
@@ -21,17 +21,19 @@
...
+			<Button v-if="!loading && showAddServerButton"
+				class="signaling-server__add-icon"
+				type="tertiary-no-background"
+				:aria-label="t('spreed', 'Add a new high-performance backend server')"
 				@click="newServer">
-				<span class="hidden-visually">{{ t('spreed', 'Add a new server') }}</span>
-			</a>
-			<span v-else-if="loading" class="icon icon-loading-small" />
+				<template #icon>
+					<Plus />
```
Comment (raimund-schluessler) [inline src/components/AdminSettings/SignalingServers.vue:34]: Same here, probably.

File: src/components/AdminSettings/SignalingServer.vue (hunk 1 above)
Comment (nickvergessen) [reply src/components/AdminSettings/SignalingServer.vue]: Right, also fixed stun and turn icons

File: src/components/AdminSettings/SignalingServer.vue [hunk 3]
```diff
@@ -38,17 +38,23 @@
 			@change="updateVerify">
 		<label :for="'verify' + index">{{ t('spreed', 'Validate SSL certificate') }}</label>
 
-		<a v-show="!loading"
-			v-tooltip.auto="t('spreed', 'Delete this server')"
```
Comment (raimund-schluessler) [inline src/components/AdminSettings/SignalingServer.vue:42]: Just a hint: `ButtonVue` has no tooltip by default, so if you want to keep it, you need to add it manually.

File: src/components/AdminSettings/SignalingServer.vue [hunk 4]
```diff
@@ -38,17 +38,23 @@
...
+			</template>
+		</Button>
 
 		<span v-if="server">{{ connectionState }}</span>
 	</div>
 </template>
 
 <script>
+import Button from '@nextcloud/vue/dist/Components/Button'
 import Tooltip from '@nextcloud/vue/dist/Directives/Tooltip'
```
Comment (raimund-schluessler) [inline src/components/AdminSettings/SignalingServer.vue:56]: Import can go, unless you want to keep the tooltip for the Button, in which case you need to add it to the button manually, see above.

File: src/components/AdminSettings/SignalingServers.vue [hunk 5]
```diff
@@ -84,7 +86,10 @@
 
 <script>
 import SignalingServer from '../../components/AdminSettings/SignalingServer.vue'
+import Button from '@nextcloud/vue/dist/Components/Button'
 import Tooltip from '@nextcloud/vue/dist/Directives/Tooltip'
```
Comment (raimund-schluessler) [inline src/components/AdminSettings/SignalingServers.vue:90]: Same here, and actually various other places where you switched to `Button`.

File: src/components/AdminSettings/SignalingServer.vue (hunk 4 above)
Comment (nickvergessen) [reply src/components/AdminSettings/SignalingServer.vue]: Removed

File: src/components/AdminSettings/SignalingServer.vue (hunk 3 above)
Comment (nickvergessen) [reply src/components/AdminSettings/SignalingServer.vue:42]: Should be fine for now if it is the accepted default behaviour. The icon should indicate it pretty obvious and screenreaders have the aria-label

File: src/components/AdminSettings/SignalingServers.vue [hunk 6]
```diff
@@ -184,6 +188,21 @@ export default {
...
+		display: flex;
+		align-items: center;
+	}
+
+	&__add-icon {
+		display: inline-block;
+		width: 44px;
+		height: 44px;
+		vertical-align: middle;
+	}
```
Comment (raimund-schluessler) [inline src/components/AdminSettings/SignalingServers.vue:203]: Is this necessary? Should be handled by the `Button`, no?

**Review by raimund-schluessler — APPROVED** [review]
Code looks good, I didn't test it though.

File: src/components/AdminSettings/SignalingServers.vue (hunk 6 above)
Comment (nickvergessen) [reply src/components/AdminSettings/SignalingServers.vue]: No, that's right. Not sure where it is coming from.

**Pull request merged**